server/routes/documents/letters-list/letters-list.controller.js

```diff
@@ -264,14 +265,40 @@
   // the same approach could be used in other places where we have checkboxes for selecting items from big lists
   module.exports.checkJuror = function(app) {
     return function(req, res) {
-      const { isChecking } = req.body;
+      const { isChecking, isCheckAll } = req.body;
 
-      if (!req.session.documentsJurorsList.checkedJurors) {
-        req.session.documentsJurorsList.checkedJurors = [];
+      if (isCheckAll === 'true') {
+        if (areAllChecked(req)) {
+          req.session.documentsJurorsList.checkedJurors = [];
+        } else {
+          req.session.documentsJurorsList.checkedJurors = [];
+          req.session.documentsJurorsList.data.forEach(juror => {
+            if (!juror[4] && !juror[5]) {
```

Comment:
these probably need to be inverted and check if they exist rather than check for not, otherwise it will never add anything to the list

Reply:
Updated